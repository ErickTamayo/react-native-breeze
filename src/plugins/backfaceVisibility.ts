import { PluginFunction, PluginPattern } from "./types";

export type PluginGroups = { visibility: "visible" | "hidden" };

export const pattern: PluginPattern = /^backface-(?<visibility>visible|hidden)$/;

export const plugin: PluginFunction<PluginGroups> = ({ groups }) => {
  const { visibility } = groups;
  return { backfaceVisibility: visibility };
};
