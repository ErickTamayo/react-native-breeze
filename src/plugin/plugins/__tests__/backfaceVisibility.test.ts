import {
  generateInput,
  shouldEvaluateTheCorrectPatternTest,
  shouldMatchOutputSnapshot,
} from "../../utils/tests";
import { pattern, plugin } from "../backfaceVisibility";

describe("backface", () => {
  const shouldMatch = generateInput("backface", ["hidden", "visible"]);

  shouldEvaluateTheCorrectPatternTest(pattern, {
    shouldMatch,
    shouldNotMatch: ["unknown"],
  });

  shouldMatchOutputSnapshot(pattern, plugin, shouldMatch);
});
