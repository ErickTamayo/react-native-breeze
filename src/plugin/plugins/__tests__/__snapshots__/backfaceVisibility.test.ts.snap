// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`backface Should parse (backface-hidden) correctly 1`] = `
Object {
  "backfaceVisibility": "hidden",
}
`;

exports[`backface Should parse (backface-visible) correctly 1`] = `
Object {
  "backfaceVisibility": "visible",
}
`;
